### PR TITLE
Add a simple logging interface that callers can implement

### PR DESCRIPTION
**Pull request by gir**
Currently this only logs the call limit header

### CODE REVIEW

**Review by johnpfeifferhelix — APPROVED** [review]
sweet
Which callers use this?

File: goshopify.go [hunk 1]
```diff
@@ -205,6 +210,12 @@ func NewClient(app App, shopName, token string) *Client {
 	return c
 }
 
+// WithLogger just sets logger and returns the Client.
+func (c *Client) WithLogger(l Logger) *Client {
```
Comment (MayaLi) [inline goshopify.go]: Another way to do this is to do WithContext(). Add the api limit info into the context which can be logged along with client information each time Do() is called

**Comment by MayaLi** [comment]
@johnpfeifferhelix according to Gigi the following repos reference this library 
```
https://github.com/myhelix/data-migration
https://github.com/myhelix/hmapi
https://github.com/myhelix/order
https://github.com/myhelix/customer
https://github.com/myhelix/fulfillment-workflow
https://github.com/myhelix/shopify-app
```

**Comment by helixjim** [comment]
@aryy @wangdrew is someone merging this in kaniel's absence? cc @johnpfeifferhelix @MayaLi

File: goshopify.go (hunk 1 above)
Comment (the-gigi) [inline goshopify.go:214]: using l (lower-case L) as a name is not recommended. too easy to confuse  with I (upper-case eye) and 1 (one) depending on the font. Not a big deal here, but still...

File: goshopify.go [hunk 2]
```diff
@@ -220,6 +231,10 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		return err
 	}
 
+	if callLimit, ok := resp.Header[callLimitHeader]; ok {
```
Comment (the-gigi) [inline goshopify.go:234]: I didn't know shopify actually sends the current bucket capacity in each response. I think we can utilize it to control the overall rate without explicit coordination between all the services that call shopify and contribute to API usage.

**Review by the-gigi — APPROVED** [review]

**Pull request merged**